src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +146,159 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
+        tasks: List[futures.Future] = []
+        with futures.ThreadPoolExecutor(
+            max_workers=self._default_thread_pool_size,
+            thread_name_prefix="import-models-to-curated-hub",
+        ) as deploy_executor:
+            for model in models_to_sync:
+                task = deploy_executor.submit(self._sync_public_model_to_hub, model)
+                tasks.append(task)
+
+        # Handle failed imports
+        results = futures.wait(tasks)
+        failed_imports: List[Dict[str, Any]] = []
+        for result in results.done:
+            exception = result.exception()
+            if exception:
+                failed_imports.append(
+                    {
+                        "Exception": exception,
+                        "Traceback": "".join(
+                            traceback.TracebackException.from_exception(exception).format()
+                        ),
+                    }
+                )
+        if failed_imports:
+            raise RuntimeError(
+                f"Failures when importing models to curated hub in parallel: {failed_imports}"
+            )
+
+    def _sync_public_model_to_hub(self, model: Dict[str, str]):
+        """Syncs a public JumpStart model version to the Hub. Runs in parallel."""
+        model_name = model["name"]
+        model_version = model["version"]
+
+        model_specs = verify_model_region_and_return_specs(
+            model_id=model_name,
+            version=model_version,
+            region=self.region,
+            scope=JumpStartScriptScope.INFERENCE,
+            sagemaker_session=self._sagemaker_session,
+        )
+        # TODO: Uncomment and implement
+        # studio_specs = self.fetch_studio_specs(model_id=model_name, version=model_version)
+        studio_specs = {}
+
+        dest_location = S3ObjectLocation(
+            bucket=self.hub_bucket_name, key=f"{model_name}/{model_version}"
+        )
+        # TODO: Validations? HeadBucket?
+
+        file_generator = FileGenerator(self.region, self._s3_client, studio_specs)
+        src_files = file_generator.format(model_specs)
+        dest_files = file_generator.format(dest_location)
+
+        files_to_copy = list(FileSync(src_files, dest_files, dest_location).call())
+
+        if len(files_to_copy) > 0:
+            # Copy files with MPU
+            print("hi")
+
+        hub_content_document = HubContentDocument_v2(spec=model_specs)
+
+        self._sagemaker_session.import_hub_content(
+            document_schema_version=HubContentDocument_v2.SCHEMA_VERSION,
+            hub_content_name=model_name,
+            hub_content_version=model_version,
+            hub_name=self.hub_name,
+            hub_content_document=hub_content_document,
+            hub_content_type=HubContentType.MODEL,
+            hub_content_display_name="",
+            hub_content_description="",
+            hub_content_markdown="",
```

Comment:
can we set to `None` instead?

Reply:
These are just placeholders

Reply:
I know, but empty string can fail validation later

Reply:
ack - but we should pass the whole content of the markdown file associated with the model here.